phpBB/phpbb/install/helper/database.php

```diff
@@ -0,0 +1,459 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\install\helper;
+
+use phpbb\install\exception\invalid_dbms_exception;
+
+/**
+ * Database related general functionality for installer
+ */
+class database
+{
+	/**
+	 * @var \phpbb\filesystem\filesystem_interface
+	 */
+	protected $filesystem;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * @var array
+	 */
+	protected $supported_dbms;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\filesystem\filesystem_interface	$filesystem			Filesystem interface
+	 * @param string									$phpbb_root_path	Path to phpBB's root
+	 */
+	public function __construct(\phpbb\filesystem\filesystem_interface $filesystem, $phpbb_root_path)
+	{
+		$this->filesystem		= $filesystem;
+		$this->phpbb_root_path	= $phpbb_root_path;
+
+		// DBMS supported by phpBB
+		$this->supported_dbms = array(
```

Comment:
Any reason why you are defining this in the constructor and not just by default for the class member?